v-next/hardhat/src/internal/cli/prompt/prompt.ts

```diff
@@ -0,0 +1,59 @@
+/**
+ * Display a confirmation prompt to the user. The prompt will be canceled if no response is given within a specified time limit.
+ * @param name Used as the key for the answer on the returned values (answers) object.
+ * @param message The message to display when the prompt is rendered in the terminal.
+ * @param timeoutMilliseconds After how much time the prompt will be cancelled if no answer is given.
+ * @returns True or false based on the user input, or undefined if the prompt times out.
+ */
+export async function confirmationPromptWithTimeout(
+  name: string,
+  message: string,
+  timeoutMilliseconds: number = 10_000,
+): Promise<boolean | undefined> {
```

Comment:
I'm not a fan of returning `| undefined`. If the timeout ends without response, can't we assume the response is `false`?

Reply:
Good question!
Let me give you a bit of context: I use undefined because later in the logic we store the consent value on file. And if we hit the timeout it means that  we did not get a true or false value about the consent from the user (which most likely is not a person but a CI if we hit 10 seconds timeout). 
So by returning undefined I can handle this specific scenario and skip saving the consent value on file (if I just receive false I could think that the user denied the telemetry consent).
But we could argue that we could save false for undefined too on the file.
Wdyt?

Reply:
Thanks for the explanation! Given that information I think we can leave it as it is.

Reply:
I think having three possible values is inherent to this API, as that's the expected behavior. `| undefined` is probably the less annoying third value in typescript.